Migration 48 failed in prod-preview

Logs of the error in `prod-preview`: https://gist.githubusercontent.com/alexeykazakov/9d3810487f4f2177a56bd4ec7010c16b/raw/73e747c9a37ffbb26d3c0ccafc68dc13caf49ec8/gistfile1.txt

The errors in `prod-preview` about the migration are:
```
{"current_version":47,"level":"info","msg":"Attempt to update DB to version 48","next_version":48,"pkg":"migration","time":"2017-04-05 21:42:53"}
Failed to obtain reader, Failed to marshal fields to JSON, json: unsupported type: migration.fn
{"err":"Failed to execute migration of step 0 of version 48: pq: could not create unique index \"iterations_name_space_id_path_unique\"\n","level":"panic","msg":"failed migration","pid":1,"time":"2017-04-05 21:42:53"}
panic: (*logrus.Entry) (0xccc320,0xc42028de00)
```

I am getting an error locally while using a db with some data loaded by my PR https://github.com/almighty/almighty-core/pull/1134
```
- FAIL: TestMigrations (0.48s)
    migration_blackbox_test.go:133: Failed to execute the migration: Failed to execute migration of step 0 of version 48: pq: syntax error
```


The error seems to be related to GetUpdatedIterationPath function 
```
t=# UPDATE iterations set path=GetUpdatedIterationPath(id,space_id,path);
ERROR:  syntax error
DETAIL:  Unexpected end of line.
CONTEXT:  PL/pgSQL function texttoltreenode(text) line 2 at SQL statement
SQL statement "select TextToLtreeNode(concat(rootiteration::text,'.',path::text))"
PL/pgSQL function getupdatediterationpath(uuid,uuid,ltree) line 15 at SQL statement```

Locally for me, the error seems to be in `TextToLtreeNode(concat(rootiteration::text,'.','path::text'))` because `path` CAN Be NULL.

--- a/migration/sql-files/048-unique-iteration-name-create-new-iteration.sql
+++ b/migration/sql-files/048-unique-iteration-name-create-new-iteration.sql
@@ -1,5 +1,14 @@
 -- Update iterations having same name, append UUID to make it unique.
-update iterations set name = name || '-' || uuid_generate_v4()  where id IN( select id from iterations where name in (select name from iterations group by name having count(name) >1 ));
+UPDATE iterations
+SET name = name || '-' || uuid_generate_v4()
+WHERE id IN
+    (SELECT id
+     FROM iterations
+     WHERE name IN
+         (SELECT name
+          FROM iterations
+          GROUP BY name
+          HAVING count(name) >1));
 
 ------  For existing spaces in production, which dont have a root iteration, create one.
 --

--- a/migration/sql-files/048-unique-iteration-name-create-new-iteration.sql
+++ b/migration/sql-files/048-unique-iteration-name-create-new-iteration.sql
@@ -1,5 +1,5 @@
------- You can't allow the same iteration name and the same ancestry inside a space
-ALTER TABLE iterations ADD CONSTRAINT iterations_name_space_id_path_unique UNIQUE(space_id,name,path);
+-- Update iterations having same name, append UUID to make it unique.
+update iterations set name = name || '-' || uuid_generate_v4()  where id IN( select id from iterations where name in (select name from iterations group by name having count(name) >1 ));
 
 ------  For existing spaces in production, which dont have a root iteration, create one.
 --
@@ -48,7 +48,10 @@ CREATE OR REPLACE FUNCTION GetUpdatedIterationPath(iteration_id uuid,space_id uu
      DECLARE
           rootiteration uuid;
      BEGIN
-     
+     If path IS NULL
+        THEN
+            path = '';
+     END IF;
      select GetRootIteration(space_id) into rootiteration;
      IF rootiteration != iteration_id 
          THEN                  
@@ -78,3 +81,6 @@ DROP FUNCTION GetRootIteration(uuid);
 DROP FUNCTION TextToLtreeNode(text);
 
 CREATE INDEX ix_name ON iterations USING btree (name);
+
+------ You can't allow the same iteration name and the same ancestry inside a space
+ALTER TABLE iterations ADD CONSTRAINT iterations_name_space_id_path_unique UNIQUE(space_id,name,path);

--- a/migration/sql-files/048-unique-iteration-name-create-new-iteration.sql
+++ b/migration/sql-files/048-unique-iteration-name-create-new-iteration.sql
@@ -48,6 +48,7 @@ CREATE OR REPLACE FUNCTION GetUpdatedIterationPath(iteration_id uuid,space_id uu
      DECLARE
           rootiteration uuid;
      BEGIN
+     -- In production this probably not NULL; safety check.
      If path IS NULL
         THEN
             path = '';

